internal/isoutil/isoutil.go

```diff
@@ -228,3 +236,69 @@ func (h *installerHandler) VolumeIdentifier() (string, error) {
 
 	return strings.TrimSpace(string(volumeId)), nil
 }
+
+func GetFileLocation(filePath, isoPath string) (int64, error) {
+	d, err := diskfs.OpenWithMode(isoPath, diskfs.ReadOnly)
+	if err != nil {
+		return 0, err
+	}
+
+	fs, err := d.GetFilesystem(0)
+	if err != nil {
+		return 0, err
+	}
+
+	isoFile, err := fs.OpenFile(filePath, os.O_RDONLY)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Failed to open file %s", filePath)
+	}
+
+	myFile := isoFile.(*iso9660.File)
+	defaultSectorSize := int64(2 * 1024)
+	offset := int64(myFile.Location()) * defaultSectorSize
+	return offset, nil
+}
+
+func WriteHeader(ignitionOffsetInfo, ramDiskOffsetInfo *OffsetInfo, isoPath string) error {
+	iso, err := os.OpenFile(isoPath, os.O_WRONLY, 0o664)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	headerOffset := int64(32768) // first 32KB in ISO
+
+	// Write ignition config
+	headerOffset, err = writeOffsetInfo(headerOffset, ignitionOffsetInfo, iso)
```

Comment:
`writeOffsetInfo` is writing stuff backwards and the offset it's receiving as a parameter is the *end* of the struct it's writing, this is very surprising and some documentation / better naming would help here

Reply:
Right, it's inspired by coreos-assembler implementation:
https://github.com/coreos/coreos-assembler/blob/510dbec7b84aa45a646079fe3341e2b5925c0774/src/cmd-buildextend-live#L510
This way we could keep some consistency and easily add more offsets later on.